woocommerce/woocommerce
# [Experimental]Attribute Filter: Dynamic title and description.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #49979.

This PR:
- Registers variation for each product attribute.
- Removes the need of manipulating metadata/blockName.
- Improves editor performance by only updating wrapper and heading on product attribute change (instead of updating every time the edit component is rendered).



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
1. Edit the Product Filters template part.
2. Reset the template part.
3. Choose the Filter Options block of the Attribute filter block, change the attribute to another one.
4. Verify the title and description of the wrapper block is updated to match the newly selected product attribute.

| Before | After |
|--------|--------|
| <img width="1243" alt="image" src="https://github.com/user-attachments/assets/8a97e35f-3ed1-439c-af1f-6d1078aa06d4"> | <img width="1243" alt="image" src="https://github.com/user-attachments/assets/e41c2105-7a4d-49cf-82b0-71535f1ca496"> |



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
